tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -410,21 +439,27 @@ private async Task DeleteAbandonedWidgetAsync(ComSafeWidget widget)
         _log.Information($"After delete, {length} widgets for this host");
     }
 
-    private async Task PinDefaultWidgetsAsync()
+    private async Task PinDefaultWidgetsAsync(CancellationToken cancellationToken)
     {
         var comSafeWidgetDefinitions = await ComSafeHelpers.GetAllOrderedComSafeWidgetDefinitions(ViewModel.WidgetHostingService);
         foreach (var comSafeWidgetDefinition in comSafeWidgetDefinitions)
         {
+            if (cancellationToken.IsCancellationRequested)

```

Comment:
I would strongly recommend changing the cancellation token usage pattern to assume it throws and to throw when cancellation is requested. Many methods that support cancellation will use CancellationToken.ThrowIfCancellationRequested(), and if this cancellation token is eventually passed to a method outside Dev Home that doesn't quietly return as this code does and instead throws, we'll have a potential unhandled exception. So I would recommend changing this pattern to use ThrowIfCancellationRequested() and assume OperationCanceledException should be caught for this scenario. We should be then free to pass the cancellation token to any APIs that use one, even outside Dev Home and can handle any throw and cleanup as appropriate with other exception handling.

Reply:
Good catch, forgot that it can throw as well 😬